site/app/controllers/forum/ForumController.php

```diff
@@ -151,6 +154,34 @@ public function publishPost(){
         }
     }
 
+    public function replyPost(){
+        $post_content = htmlentities($_POST["reply_post_content"], ENT_QUOTES | ENT_HTML5, 'UTF-8');
+        $thread_id = htmlentities($_POST["reply_thread_id"], ENT_QUOTES | ENT_HTML5, 'UTF-8');
+        $parent_id = htmlentities($_POST["reply_parent_id"], ENT_QUOTES | ENT_HTML5, 'UTF-8');
+        $anon = (isset($_POST["Anon"]) && $_POST["Anon"] == "Anon") ? 1 : 0;
+        if(empty($post_content) || empty($thread_id)){
+            $this->core->addErrorMessage("There was an error with your reply.");
+            $this->core->redirect($this->core->buildUrl(array('component' => 'forum', 'page' => 'view_thread')));
+        } else {
+            // $hasGoodAttachment = $this->checkGoodAttachment(false, $_POST["post_content"], $thread_id);
+            // if($hasGoodAttachment == -1){
+            //     return;
+            // }
+            
+            $post_id = $this->core->getQueries()->createReply($this->core->getUser()->getId(), $post_content, $thread_id, $anon, 0, $parent_id, false);
+            $thread_dir = FileUtils::joinPaths(FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "forum_attachments"), $thread_id);
+            // if($hasGoodAttachment == 1) {
+            //     $post_dir = FileUtils::joinPaths($thread_dir, $post_id);
+            //     FileUtils::createDir($post_dir);
+            //     for($i = 0; $i < count($_FILES["file_input"]["name"]); $i++){
+            //         $target_file = $post_dir . "/" . basename($_FILES["file_input"]["name"][$i]);
+            //         move_uploaded_file($_FILES["file_input"]["tmp_name"][$i], $target_file);
+            //     }
+            // }
+            $this->core->redirect($this->core->buildUrl(array('component' => 'forum', 'page' => 'view_thread', 'thread_id' => $thread_id)));
+        }
+    }
+
```

Comment:
This section of code seems to be a duplicate of publishPost with attachment code commented out. Can you use the publishPost function but just send the parent_id as a POST to publishPost similar to how you are doing in this function.

Reply:
Will do